components/newsletter-sign-up/newsletter-sign-up.jsx

```diff
@@ -0,0 +1,288 @@
+import React from "react";
+import ReactGA from "react-ga";
+import classNames from "classnames";
+import Service from "../../js/service";
+
+class NewsletterSignUp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      apiSubmitted: false,
+      apiSuccess: false,
+      apiFailed: false,
+      userTriedSubmitting: false
+    };
+  }
+
+  /**
+   * Ensure that the parent component is informed
+   * about this component being mounted (primarily
+   * used in the context of automated testing)
+   */
+  componentDidMount() {
+    if (this.props.whenLoaded) {
+      this.props.whenLoaded();
+    }
+  }
+
+  // state update function
+  apiSubmissionSuccessful() {
+    this.setState({
+      apiSuccess: true
+    });
+  }
+
+  // state update function
+  apiSubmissionFailure(e) {
+    if (e && e instanceof Error) {
+      this.setState({
+        apiFailed: true
+      });
+    }
+  }
+
+  // state check function
+  apiIsDone() {
+    return (
+      this.state.apiSubmitted && (this.state.apiSuccess || this.state.apiFailed)
+    );
+  }
+
+  /**
+   * Submit the user's data to the API server.
+   */
+  submitDataToApi() {
```

Comment:
Utilizing Pulse's existing `service.js` instead of https://github.com/mozilla/foundation.mozilla.org/blob/master/source/js/components/join/join.jsx#L51-L93